### PR TITLE
calc logic and css

### PR DESCRIPTION
**Pull request by pacynek93**

### CODE REVIEW

File: src/components/Buttons/Buttons.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      <div className="btn" onClick={() => handleButtonClick(".")}>
+        .
+      </div>
+      <div
+        className="btn"
+        data-type="operator"
+        onClick={() => handleButtonClick("=")}
+      >
+        =
+      </div>
```
Comment (DMTcorp) [inline src/components/Buttons/Buttons.js:196]: Use `Array.map()` here

File: src/components/Screen/Screen.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  const updateScreen = (value) => {
+    setScreenContent(value);
+  };
+
+  return (
+    <>
+      <div className="screen">
+        <div>{screenContent}</div>
+      </div>
+      <Buttons updateScreen={updateScreen} />
```
Comment (DMTcorp) [inline src/components/Screen/Screen.js:17]: you can directly pass `setScreenContent` here

**Review by DMTcorp — CHANGES_REQUESTED** [review]

File: src/components/Buttons/Buttons.js
Comment (oskarkasprzyk) [inline src/components/Buttons/Buttons.js:1]: Why use div selector instead button selector?
You use"operator" and "special" as css class but you didnt define it

**Comment by DMTcorp** [comment]
1. The structure is a bit confusing, I would suggest something like that:
```
App
-- Screen (or Display)
-- Keyboard
-- -- Button 
```

2. I would also suggest to extract the calc logic to a different file
3. Use [Context](https://legacy.reactjs.org/docs/context.html) to avoid prop drilling

File: src/components/Button/Button.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+
+const Button = ({ button, handleButtonClick }) => (
+  <button
+    className={`btn ${button.className}`}
+    onClick={() => handleButtonClick(button.value)}
+  >
+    {button.value === "/" ? "÷" : button.value === "*" ? "x" : button.value}
```
Comment (oskarkasprzyk) [inline src/components/Button/Button.js:8]: Don't nest ternary expressions!

File: src/components/Button/Button.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+
+const Button = ({ button, handleButtonClick }) => (
```
Comment (oskarkasprzyk) [inline src/components/Button/Button.js:3]: If you don't use prop-types or typescript, I would prefer remove button prop and make onClick + children + className props, it's more readable for other

File: src/components/Keyboard/Keyboard.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import "./Keyboard.css";
+import Button from "../Button/Button";
+
+const Keyboard = ({ handleButtonClick }) => {
+  const buttonValues = [
+    { value: "AC", className: "special" },
+    { value: "%", className: "special" },
+    { value: "+/-", className: "special" },
+    { value: "/", className: "operator" },
+    { value: "7", className: "" },
```
Comment (oskarkasprzyk) [inline src/components/Keyboard/Keyboard.js:11]: Remove from array keys with empty value

File: src/components/Buttons/Buttons.js
Comment (pacynek93) [reply src/components/Buttons/Buttons.js:1]: done

File: src/components/Button/Button.js (hunk 3 above)
Comment (pacynek93) [reply src/components/Button/Button.js]: Done, moved to Keyboard.js

File: src/components/Button/Button.js (hunk 4 above)
Comment (pacynek93) [reply src/components/Button/Button.js]: done

File: src/components/Wrapper/Wrapper.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import "./Wrapper.css";
+
+const Wrapper = ({ children }) => {
+  return <div className="wrapper"> {children} </div>;
+};
+
+export default Wrapper;
```
[the remark this answers is not in the document]
Comment (pacynek93) [reply src/components/Wrapper/Wrapper.js]: Correct, component deleted

File: src/components/Keyboard/Keyboard.js [hunk 7]
```diff
@@ -8,32 +8,48 @@ const Keyboard = ({ handleButtonClick }) => {
...
     <>
-      {buttonValues.map((button, index) => (
-        <Button
-          key={index}
-          button={button}
-          handleButtonClick={handleButtonClick}
-        />
-      ))}
+      {buttonValues.map((button, index) => {
+        let buttonText;
```
Comment (oskarkasprzyk) [inline src/components/Keyboard/Keyboard.js:31]: Remove switch and make object like
`const MathOperatorMath = {
  '/': '÷',
  ...rest
}`

Comment (pacynek93) [reply src/components/Keyboard/Keyboard.js]: Ok refactored in last commit

**Pull request merged**